### PR TITLE
Add property to provide a http proxy url to connect to glue registry

### PR DESCRIPTION
**Pull request by abhilash47**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java [hunk 1]
```diff
@@ -90,11 +91,26 @@ public AWSSchemaRegistryClient(@NonNull AwsCredentialsProvider credentialsProvid
...
                 .build();
 
+        UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder = UrlConnectionHttpClient.builder();
+
+        if (glueSchemaRegistryConfiguration.getProxyUrl() != null) {
+        	log.debug("Creating http client using proxy {}", glueSchemaRegistryConfiguration.getProxyUrl());
+        	try {
+    	    	URI uri = new URI(glueSchemaRegistryConfiguration.getProxyUrl());
+    	        ProxyConfiguration proxy = ProxyConfiguration.builder().endpoint(uri).build();
+    	        urlConnectionHttpClientBuilder.proxyConfiguration(proxy);
```
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:101]: The indentation seems off, please fix it.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java [hunk 2]
```diff
@@ -90,11 +91,26 @@ public AWSSchemaRegistryClient(@NonNull AwsCredentialsProvider credentialsProvid
...
+        	} catch (URISyntaxException e) {
+        		String message = String.format("Malformed uri, please pass the valid proxy uri for creating the client",
+                        glueSchemaRegistryConfiguration.getEndPoint());
```
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:104]: This error message seems incorrect. Should this be `.getProxyUrl` instead? Also, the `String.format` parameters must have a placeholder `%s' for printing the Uri.

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java [hunk 3]
```diff
@@ -18,6 +18,10 @@
 import software.amazon.awssdk.services.glue.model.Compatibility;
 
 public final class AWSSchemaRegistryConstants {
+	/**
```
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java:21]: Indentation seems off.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java [hunk 4]
```diff
@@ -90,11 +91,26 @@ public AWSSchemaRegistryClient(@NonNull AwsCredentialsProvider credentialsProvid
...
 
+        UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder = UrlConnectionHttpClient.builder();
+
+        if (glueSchemaRegistryConfiguration.getProxyUrl() != null) {
+        	log.debug("Creating http client using proxy {}", glueSchemaRegistryConfiguration.getProxyUrl());
+        	try {
+    	    	URI uri = new URI(glueSchemaRegistryConfiguration.getProxyUrl());
+    	        ProxyConfiguration proxy = ProxyConfiguration.builder().endpoint(uri).build();
+    	        urlConnectionHttpClientBuilder.proxyConfiguration(proxy);
+        	} catch (URISyntaxException e) {
```
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:102]: You might want to consider moving this to validation to `GlueSchemaRegistryConfiguration.java` similar to how it is being done for the configuration parameters.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java [hunk 5]
```diff
@@ -90,11 +91,26 @@ public AWSSchemaRegistryClient(@NonNull AwsCredentialsProvider credentialsProvid
                 .addExecutionInterceptor(new UserAgentRequestInterceptor())
                 .build();
 
+        UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder = UrlConnectionHttpClient.builder();
```
[the remark this answers is not in the document]
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:93]: @blacktooth  - Added unit tests

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java (hunk 4 above)
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: @blacktooth  moved the validation to GlueSchemaRegistryConfiguration.java

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java (hunk 1 above)
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: fixed the indentation

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java (hunk 2 above)
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: fixed it

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java (hunk 3 above)
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java]: fixed indentation

**Comment by abhilash47** [comment]
@blacktooth - I am not able to reproduce the failing windows builds on my windows laptop. Could you please help me with whats causing the failure ?

**Comment by blacktooth** [comment]
Thanks for checking! It looks like the Windows builds are failing irrespective of this change. We will look into those separately. This PR can be merged after approval.

**Comment by abhilash47** [comment]
@blacktooth - I have pulled the latest changes to resolve windows build issues and version upgrades. Could this PR be approved and merged ?

**Comment by YangSan0622** [comment]
@abhilash47 Hello, we are working on our internal process for this PR, we will update you once that process is completed.

**Comment by YangSan0622** [comment]
@abhilash47 Hello, we have got the approval needed for this PR, we will merge it shortly. Thanks a lot!

**Review by blacktooth — APPROVED** [review]

**Pull request merged**